### DISCUSSION TITLE
Disable Github Actions check runs from annotating files

### DISCUSSION THREAD

**Discussion by tamdilip** [community/community#25336] [Actions]
<p>Looking for a way to disable Github Actions check runs from annotating files.</p>
<p>/* Context -  Working on a simple eslint workflow action to comment on PR’s, as it’s annoying with this check runs annotating all files by default */</p>
<p>Ref PR- <a href="https://github.com/tamdilip/ember_poc/pull/143/files" rel="nofollow noopener">https://github.com/tamdilip/ember_poc/pull/143/files</a></p>

**Comment by Yanjingzhu** [comment]
<p>I checked your PR and workflow yaml, the annotations are added by action in “tamdilip/eslint-review-action” repo.<br>
What’s the source code of those action? Would you mind changing the repo to a public one and share its link here?</p>

**Comment by tamdilip** [comment]
<p>Thanks for looking into this <a class="mention" href="/yanjingzhu">@yanjingzhu</a>, I have made the action repo public as you requested and this is the current branch I’m testing with - <a href="https://github.com/tamdilip/eslint-review-action/tree/feature/cache-and-npm-check" rel="nofollow noopener">https://github.com/tamdilip/eslint-review-action/tree/feature/cache-and-npm-check</a></p>

**Comment by Yanjingzhu** [comment]
<p>I checked the index.js file in the action repo. It invoke “<a href="https://developer.github.com/v3/checks/runs/#update-a-check-run" rel="nofollow noopener">Update a check run</a>” API .<br>
There are many annotations in the check run . These annotations will be shown in the related files.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7a4d4bdd668e51d5b135c86521df77874545dace.png" data-download-href="/uploads/short-url/hrVRcmqMPhdCAfIxkq3ySh1O0AS.png?dl=1" title="annotation" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181090488-c77b3cb7-dc04-4b33-a7ca-5add0f1ca27e.png" alt="annotation" data-base62-sha1="hrVRcmqMPhdCAfIxkq3ySh1O0AS" width="690" height="326" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7a4d4bdd668e51d5b135c86521df77874545dace_2_690x326.png, https://aws1.discourse-cdn.com/github/optimized/2X/7/7a4d4bdd668e51d5b135c86521df77874545dace_2_1035x489.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/7/7a4d4bdd668e51d5b135c86521df77874545dace_2_1380x652.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7a4d4bdd668e51d5b135c86521df77874545dace_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">annotation</span><span class="informations">1861×880 46.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
I am afraid that the annotations shown in files could not be disabled.

**Comment by tamdilip** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> :worried: I’m nowhere  explicitly invoking any check-run API call, I’m making quite other github API calls via OCTOKIT library for listing files changed in PR, existing PR comments and so on but not check-runs. I’m not sure if any of these other API calls are internally invoking check-runs :face_with_head_bandage:.</p>
<p>Still it would be much helpful if we can control all these check-runs by toggling in the action.yml with a key :neutral_face: :sweat:.</p>
<p>Also if some other way to control <code>checks:write</code> permission on repo level settings atleast.</p>

**Comment by tamdilip** [comment]
<p>I tried removing the custom action repo from my <a href="https://github.com/tamdilip/ember_poc/actions/runs/138315727/workflow" rel="nofollow noopener">workflow yml</a> and just ran a test command.<br>
Still the same case :face_with_head_bandage: .</p>
<p>I could see this is being happened automatically as mentioned here - <a href="https://developer.github.com/changes/2018-05-07-new-checks-api-public-beta/" rel="nofollow noopener">https://developer.github.com/changes/2018-05-07-new-checks-api-public-beta/</a></p>

**Comment by tamdilip** [comment, marked as answer]
<p>Finally, for time being I managed to stop the annotations by capturing those <a href="https://github.com/tamdilip/eslint-review-action/blob/feature/cache-and-npm-check/src/command-executor.js#L47" rel="nofollow noopener">CLI logs output</a> as XML format separately via a <a href="https://github.com/tamdilip/eslint-review-action/blob/feature/cache-and-npm-check/src/command-executor.js#L35" rel="nofollow noopener">listener</a>.</p>
